pkg/plugin/validator_test.go

```diff
@@ -0,0 +1,252 @@
+			},
+			fakeTokenServerResqCode: http.StatusUnauthorized,
+			wantErrSubstr:           "failed to get access token",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "issue_not_open",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "issue is in state: closed",
+			issueBytes:              []byte(`{"state": "closed"}`),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		ctx := context.Background()
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			fakeTokenServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("Accept") != "application/vnd.github+json" {
+					w.WriteHeader(500)
+					fmt.Fprintf(w, "missing accept header")
+					return
+				}
+				authHeader := r.Header.Get("Authorization")
+				if authHeader == "" || !strings.HasPrefix(authHeader, "Bearer ") {
+					w.WriteHeader(500)
+					fmt.Fprintf(w, "missing or malformed authorization header")
+					return
+				}
+				w.WriteHeader(tc.fakeTokenServerResqCode)
+				fmt.Fprintf(w, `{"token":"this-is-the-token-from-github"}`)
+			}))
+
+			hc, done := newTestServer(testHandleIssueReturn(t, tc.issueBytes))
+			defer done()
+			ghAppOpts := []githubapp.ConfigOption{
+				githubapp.WithJWTTokenCaching(1 * time.Minute),
+				githubapp.WithAccessTokenURLPattern(fakeTokenServer.URL + "/%s/access_tokens"),
+			}
+			testGHAppCfg := githubapp.NewConfig(tc.cfg.GitHubAppID, tc.cfg.GitHubAppInstallationID, testPrivateKey, ghAppOpts...)
+			testGituhbApp := githubapp.New(testGHAppCfg)
+
+			validator, err := NewValidator(tc.cfg, WithGitHubClient(github.NewClient(hc)), WithGithubApp(testGituhbApp))
+			if err != nil {
+				t.Fatalf("failed to create validator: %v", err)
+			}
+			gotErr := validator.MatchIssue(ctx, tc.issueURL)
+			if diff := testutil.DiffErrString(gotErr, tc.wantErrSubstr); diff != "" {
+				t.Errorf("Process(%+v) got unexpected error substring: %v", tc.name, diff)
+			}
+		})
+	}
+}
+
+func testGeneratePrivateKey(tb testing.TB) (string, *rsa.PrivateKey) {
+	tb.Helper()
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		tb.Fatalf("Error generating RSA private key: %v", err)
+	}
+
+	// Encode the private key to the PEM format
+	privateKeyPEM := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
+	}
+	buf := new(bytes.Buffer)
+	err = pem.Encode(buf, privateKeyPEM)
+	if err != nil {
+		tb.Fatalf("Error encoding privateKeyPEM: %v", err)
+	}
+	return buf.String(), privateKey
+}
+
+// newTestServer creates a fake http client.
+func newTestServer(handler func(w http.ResponseWriter, r *http.Request)) (*http.Client, func()) {
```

Comment:
Pass in the `t *testing.T` and use `t.Cleanup()` to close resources

Reply:
I think `newTestServer` returns the following function as `done()`, and the resources are closed by `defer done()`.
```
func() {
	tr.CloseIdleConnections()
	ts.Close()
}
```

Reply:
Yes, I saw that. But it's better to use `t.Cleanup()` in tests

Reply:
Done.